src/pages/aspect/AspectSettingsPage.tsx

```diff
@@ -42,6 +46,30 @@ const AspectSettingsPage: FC<AspectSettingsPageProps> = ({ paths: _paths }) => {
     >
       {(entities, state, actions) => {
         const visuals = (entities.aspect ? [entities.aspect.banner, entities.aspect.bannerNarrow] : []) as Visual[];
+        const btnDisabled = !aspect || !entities.aspect || state.updating || state.deleting;
+
+        const handleDelete = async () => {
+          if (!entities.aspect || !aspect) {
+            return;
+          }
+
+          await actions.handleDelete(entities.aspect.id);
```

Comment:
Should we `await`? If deletion takes a few seconds, a user may already start interacting - then he would be suddenly taken away from the page - which is suboptimal UX.

Reply:
When the deletion takes a few seconds, and the user is navigated to a screen which will refetch the same data some errors are thrown because you tried to fetch the entity and it's sub-entities in the process and they may not exist.
I am opened for other suggestions.

Reply:
Are you sure the errors are thrown? Because I assume most frameworks start cascaded deletion from the topmost entity.

Reply:
Otherwise any other user who opened the same page while some deletion is being performed would get the same errors 🤷‍♂️